packages/flutter/test/widgets/router_test.dart

```diff
@@ -1032,6 +1032,64 @@ testWidgets('ChildBackButtonDispatcher take priority recursively', (WidgetTester
     await tester.pump();
     expect(find.text('popped inner1'), findsOneWidget);
   });
+
+  testWidgets('`didUpdateWidget` test', (WidgetTester tester) async {
```

Comment:
The existing  [test ](https://github.com/douglasiacovelli/flutter/blob/bd4043077de1a7f437778d45076ddf00d941f25b/packages/flutter/test/widgets/router_test.dart#L777) case will trigger `didUpdateWidget` and `didChangeDependencies`(`_RouterScope` InheritedWidget change) at the same time, so no bug will be triggered.